### ISSUE TITLE
Releasing code to use DinoV2 to calculate Feature loss.

### ISSUE THREAD

**Issue by BurguerJohn** [facebookresearch/dinov2#76]
This code is designed for my personal projects, where I use VGG as the backbone. However, I have also conducted a few tests using DinoV2 as the backbone, and the results have been quite good. Although there may be room for further improvement, it already performs really well for my own projects.

[https://github.com/BurguerJohn/torch-felix/blob/main/example_dinoV2_loss.py](https://github.com/BurguerJohn/torch-felix/blob/main/example_dinoV2_loss.py)

You can learn more about the code here:
[https://github.com/BurguerJohn/torch-felix](https://github.com/BurguerJohn/torch-felix)

My projects typically require all layers of the pre-trained model to function optimally, so you might need to modify the code to suit your specific needs. Nevertheless, I hope this serves as a helpful starting point.

**Comment by woctezuma** [comment]
I cannot say that I understand much, but based on the code and the README, the code is used to compute a distance between features, and this distance would be an L2 loss between the concatenated outputs of all the layers.

**Comment by BurguerJohn** [comment]
Sorry, I will try to update the readme later to explain better, but it is pretty much this paper with some improvements and flexibility that is easy to plug in Dino and other models:
https://paperswithcode.com/method/vgg-loss

Usually used for super resolution loss and models that output some image.
